apps/coscrad-frontend/src/components/resources/spatial-features/leaflet/spatial-feature-marker.tsx

```diff
@@ -0,0 +1,98 @@
+import { GeometricFeatureType, ISpatialFeatureViewModel } from '@coscrad/api-interfaces';
+import { Icon as LeafletIcon, Marker as LeafletMarker } from 'leaflet';
+import iconUrl from 'leaflet/dist/images/marker-icon.png';
+import shadowUrl from 'leaflet/dist/images/marker-shadow.png';
+import {
+    Marker as PointMarker,
+    Polygon as PolygonMarker,
+    Polyline as PolylineMarker,
+    Popup as LeafletPopup,
+} from 'react-leaflet';
+import { SinglePropertyPresenter } from '../../../../utils/generic-components';
+import { SpatialFeatureDetailPresenter } from '../map';
+
+interface MarkerPresenterProps<T = unknown> {
+    spatialFeature: ISpatialFeatureViewModel<T>;
+    DetailPresenter: SpatialFeatureDetailPresenter;
+}
+
+const lookupTable: {
+    [K in GeometricFeatureType]: (props: MarkerPresenterProps) => JSX.Element;
+} = {
+    [GeometricFeatureType.point]: ({
+        spatialFeature,
+        DetailPresenter,
+    }: MarkerPresenterProps<[number, number]>) => (
+        <PointMarker
+            key={spatialFeature.id}
+            position={spatialFeature.geometry.coordinates as [number, number]}
+        >
+            <LeafletPopup>
+                <SinglePropertyPresenter display="ID" value={spatialFeature.id} />
+                <DetailPresenter {...spatialFeature} />
+                {/* <PointPresenter coordinates={coordinates} /> */}
+            </LeafletPopup>
+        </PointMarker>
+    ),
+    [GeometricFeatureType.line]: ({
+        spatialFeature,
+        DetailPresenter,
+    }: MarkerPresenterProps<[number, number][]>) => (
+        <PolylineMarker positions={spatialFeature.geometry.coordinates}>
+            <LeafletPopup>
+                <SinglePropertyPresenter display="ID" value={spatialFeature.id} />
+                <DetailPresenter {...spatialFeature} />
+            </LeafletPopup>
+        </PolylineMarker>
+    ),
+    [GeometricFeatureType.polygon]: ({
+        spatialFeature,
+        DetailPresenter,
+    }: MarkerPresenterProps<[number, number][][]>) => (
+        <PolygonMarker positions={spatialFeature.geometry.coordinates}>
+            <LeafletPopup>
+                <SinglePropertyPresenter display="ID" value={spatialFeature.id} />
+                <DetailPresenter {...spatialFeature} />
+                {/* <PolygonPresenter coordinates={coordinates} /> */}
+            </LeafletPopup>
+        </PolygonMarker>
+    ),
+};
+
+/**
+ * Note that we refer to any element that renders a spatial feature on a map
+ * as a `Marker`. This is not exactly in keeping with Leaflet's terminology.
+ */
+export const buildSpatialFeatureMarker =
+    (DetailPresenter: SpatialFeatureDetailPresenter) =>
```

Comment:
Currying the function here?  I still need to do a tutorial on this pattern...

Reply:
Yes. Essentially, we are abstracting the differences between the markers for different spatial feature types so it isn't necessary to know about them outside of here.